src/RecyclableMemoryStream.cs

```diff
@@ -551,6 +551,14 @@ public override byte[] GetBuffer()
             return this.largeBuffer;
         }
 
+#if NETCOREAPP2_1_OR_GREATER || NETSTANDARD2_1
+        /// <inheritdoc/>
+        public override void CopyTo(Stream destination, int bufferSize)
+        {
+            CopyToAsync(destination, bufferSize, CancellationToken.None).GetAwaiter().GetResult();
```

Comment:
Sync over async is an antipattern. Let's reuse `WriteTo`.
```suggestion
            WriteTo(destination, this.position, this.length - this.position);
```